### PR TITLE
Output PR URL in agreement status if we know it

### LINKED ISSUE #21: Make "Details" in pull requests pre-fill out the "sync with GitHub" field
E.g. in https://github.com/whatwg/html/pull/3512 if you expand "View Details" then click "Details" next to the Participation line, you're taken to https://participate.whatwg.org/agreement-status?user=annevk&repo=html.

Currently if you want to sync that PR, you have to switch tabs back to the PR, copy the URL from the address bar, and paste it into the text field in the participate.whatwg.org page.

We should pass the PR number in the query string, and use that to pre-fill the textbox, so that you can just click "Update" without the switch-tabs-copy-paste workflow.

### PR DESCRIPTION
**Pull request by annevk**
Closes #21.

All existing tests pass which is a bit suspicious. Curious if this generally looks good. And help with tests appreciated.

### CODE REVIEW

**Comment by domenic** [comment]
Code looks pretty reasonable; it was just the threading through everything that made me avoid this for so long, so thanks for doing that!

The tests are all passing because all calls in `__tests__/get-user-status.js` are of the form

```js
  expect(await getUserStatus("johndoetw", "console")).toMatchSnapshot();
```

In particular, they don't pass the new third pull request number argument. And, your code in `lib/get-user-status.js` doesn't append `&pull=...` unless that argument is passed. This suggests maybe we don't want that `if` statement? All call sites pass in the PR number now, so I think it'd be best to just assume it's always passed in (i.e. remove the `if` wrapper).

Once you remove that, then you'll get failing tests, because it'll be expected the URL without `&pull=...`, but the new result will have `&pull=undefined`. So then, update all the tests to pass in some static PR number. Then all the tests will fail because they're expecting no `&pull=...`, but the new result is `&pull=5` or whatever.

Then you get to experience the magic of [snapshot testing](https://jestjs.io/docs/en/snapshot-testing). Run `npm test -- --updateSnapshot`. This will change the file in `__snapshots__/get-user-status.js.snap` to reflect the new results. Do a manual code review of that file to make sure the results are what you expect, and commit it. Then further runs of `npm test` will pass, by checking themselves against the updated snapshot.

**Comment by annevk** [comment]
Yeah, that might be reasonable. I don't think we want to require `pull` to be a required parameter though, do we? It seems like checking if a user signed up for a standard is still a reasonable use case.

**Comment by domenic** [comment]
Hmm, yeah, I guess allowing people to manually navigate to such URLs is probably wise, even if we never surface them.

So then we could either:

- Duplicate all the tests, one with the parameter and one without
- Keep tests mostly as-is; add a few with the PR parameter
- Update all tests to use the PR parameter, then add a few without

Any of these seems reasonable to me, although I have slight preference against duplication.

Oh, also, be sure to test an XSS case, like `getUserStatus("johndoetw", "console", "<script>alert(1);</script>")`.

**Comment by annevk** [comment]
`package-lock.json` also changed locally, but I'm not sure why so I kept it out of this changeset.

File: __tests__/__snapshots__/get-user-status.js.snap [hunk 1]
```diff
@@ -22,6 +33,17 @@ Object {
...
 `;
 
+exports[`Individual exists, but their username is spelled with a different case 2`] = `
+Object {
+  "context": "Participation",
+  "description": "@JANEdoetw has signed up to participate as an individual",
+  "isNothing": false,
+  "longDescription": "@JANEdoetw has signed up to participate as an individual. All is well; contribute at will!",
+  "state": "success",
+  "target_url": "https://participate.whatwg.org/agreement-status?user=JANEdoetw&repo=console",
```
Comment (domenic) [inline __tests__/__snapshots__/get-user-status.js.snap:43]: Why is this missing the `pull`?

File: __tests__/__snapshots__/get-user-status.js.snap [hunk 2]
```diff
@@ -110,6 +209,17 @@ Object {
...
 `;
 
+exports[`Individuals exist, but the user is not one of them; it is an XSS attempt 2`] = `
+Object {
+  "context": "Participation",
+  "description": "@<script>alert(1);</script> has not signed up to participate",
+  "isNothing": true,
+  "longDescription": "@&lt;script&gt;alert(1);&lt;/script&gt; does not appear to have signed the agreement, or be associated with an entity that has done so. Please sign it, or forward it to an authorized representative!",
+  "state": "pending",
+  "target_url": "https://participate.whatwg.org/agreement-status?user=<script>alert(1);</script>&repo=console",
```
Comment (domenic) [inline __tests__/__snapshots__/get-user-status.js.snap:219]: Ditto

**Review by domenic — APPROVED** [review]

**Pull request merged**